### PR TITLE
Fix clippy warning

### PR DESCRIPTION
**Pull request by hulthe**
This PR fixes a clippy warning that was tripping up our CI

### CODE REVIEW

**Comment by hulthe** [comment]
@faern @pylls

**Review by faern — APPROVED** [review]
Really nice! Yeah, less indexing and more iterators is really good.

**Comment by faern** [comment]
I won't merge this, at least not until @pylls give a thumbs up. I don't want to mess around in the main `maybenot` crate code without your approval :)

**Pull request merged**
